consumer/attribute_checker.go

```diff
@@ -0,0 +1,36 @@
+package consumer
+
+import (
+	"encoding/json"
+)
+
+// AttributeChecker is an interface for checking if an attribute is empty.
+type AttributeChecker interface {
+	IsEmpty() bool
+}
+
+// JSONAttributeChecker is an implementation of Checker for JSON data.
+type JSONAttributeChecker struct {
+	data []byte
+}
+
+// IsEmpty returns wether the data of the JSON data is empty
+func (j *JSONAttributeChecker) IsEmpty() bool {
+	if j.data == nil {
+		return true
+	}
+
+	var jsonData interface{}
+	if err := json.Unmarshal(j.data, &jsonData); err != nil {
+		return false
+	}
+
+	switch v := jsonData.(type) {
+	case []interface{}:
+		return len(v) == 0
+	case map[string]interface{}:
+		return len(v) == 0
```

Comment:
I guess you can group both conditions, as the return value is the same.

case []interface{}, map[string]interface{}:
    return len(v)

Reply:
turns out we can't. it fails to compile with "./attribute_checker.go:30:14: invalid argument: v (variable of type interface{}) for len
"